libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/AuthenticationUtilities.kt

```diff
@@ -94,11 +99,55 @@ internal suspend fun onAuthFlowComplete(
     buildAccountName: (username: String?, instanceServer: String?) -> String = ::defaultBuildAccountName,
     nativeLogin: Boolean = false,
 ) {
-    val context = SalesforceSDKManager.getInstance().appContext
-    val userAccountManager = SalesforceSDKManager.getInstance().userAccountManager
     val blockIntegrationUser = SalesforceSDKManager.getInstance().shouldBlockSalesforceIntegrationUser &&
             fetchIsSalesforceIntegrationUser(tokenResponse, loginServer)
 
+    onAuthFlowComplete(
+        tokenResponse,
+        loginServer,
+        consumerKey,
+        onAuthFlowError,
+        onAuthFlowSuccess,
+        buildAccountName,
+        nativeLogin,
+        SalesforceSDKManager.getInstance().appContext,
+        SalesforceSDKManager.getInstance().userAccountManager,
+        blockIntegrationUser,
+        getRuntimeConfig(SalesforceSDKManager.getInstance().appContext),
+        ::updateLoggingPrefsHelper,
+        ::fetchUserIdentity,
+        ::startMainActivityHelper,
+        ::setAdministratorPreferences,
+        ::addAccountHelper,
+        ::handleScreenLockPolicy,
+        ::handleBiometricAuthPolicy,
+        ::handleDuplicateUserAccount
+    )
+}
+
+@VisibleForTesting
+internal suspend fun onAuthFlowComplete(
```

Comment:
This is like the original onAuthFlowComplete except with pieces refactored out in private helper methods - which are not called directly but are called through lambda parameters.